api/git/git.go

```diff
@@ -1,38 +1,54 @@
 package git
 
 import (
+	"crypto/tls"
+	"net/http"
 	"net/url"
 	"strings"
+	"time"
 
 	"gopkg.in/src-d/go-git.v4"
 	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/client"
+	githttp "gopkg.in/src-d/go-git.v4/plumbing/transport/http"
 )
 
 // Service represents a service for managing Git.
-type Service struct{}
+type Service struct {
+	httpsCli *http.Client
+}
 
 // NewService initializes a new service.
 func NewService(dataStorePath string) (*Service, error) {
-	service := &Service{}
+	service := &Service{
+		&http.Client{
```

Comment:
please use the names of the fields you want to assign in any struct creation, clearer to read. In my opinion, it's also clearer to create the component outside of the service declaration:

```
httpsCli := &http.Client{...}
return &Service{
 httpsCli: httpsCli,
}
```